### PR TITLE
create folder and file structure for server

### PR DESCRIPTION
**Pull request by AhmedQeshta**
# Create File and Folder Structure
> The first is to create Folders, the second is then create files inside them.

### Server Folder and File

```
    - index.ts
    - app.js
    - test
       - database.test.ts
       - routes.test.ts
    - middlewares
        - index.ts
    - routes
        - index.ts
    - controllers
        - index.ts
    - utils
        - validation
        - Customerr.ts
        - jwt.ts
        - index.ts
    - database    
        - models
            - index.ts
        - config
           - build.sql
           - build.ts
           - connection.ts
        - fakedata   
           - index.ts
```

### CODE REVIEW

**Review by AFahgan — APPROVED** [review]
Great Job

**Review by rawaneyad — APPROVED** [review]
Great Job 👍🏼

**Review by rawaneyad — APPROVED** [review]

**Review by AFahgan — APPROVED** [review]
Great Job !

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    // "jsxFactory": "",                                 /* Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h' */
+    // "jsxFragmentFactory": "",                         /* Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'. */
+    // "jsxImportSource": "",                            /* Specify module specifier used to import the JSX factory functions when using `jsx: react-jsx*`.` */
+    // "reactNamespace": "",                             /* Specify the object invoked for `createElement`. This only applies when targeting `react` JSX emit. */
+    // "noLib": true,                                    /* Disable including any library files, including the default lib.d.ts. */
+    // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
+
+    /* Modules */
+    "module": "commonjs",                                /* Specify what module code is generated. */
+    // "rootDir": "./",                                  /* Specify the root folder within your source files. */
```
Comment (LinaYahya) [inline tsconfig.json:28]:
```suggestion
    "rootDir": "./server",                                  /* Specify the root folder within your source files. */

```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    // "checkJs": true,                                  /* Enable error reporting in type-checked JavaScript files. */
+    // "maxNodeModuleJsDepth": 1,                        /* Specify the maximum folder depth used for checking JavaScript files from `node_modules`. Only applicable with `allowJs`. */
+
+    /* Emit */
+    // "declaration": true,                              /* Generate .d.ts files from TypeScript and JavaScript files in your project. */
+    // "declarationMap": true,                           /* Create sourcemaps for d.ts files. */
+    // "emitDeclarationOnly": true,                      /* Only output d.ts files and not JavaScript files. */
+    // "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
+    // "outFile": "./",                                  /* Specify a file that bundles all outputs into one JavaScript file. If `declaration` is true, also designates a file that bundles all .d.ts output. */
+    // "outDir": "./",                                   /* Specify an output folder for all emitted files. */
```
Comment (LinaYahya) [inline tsconfig.json:50]:
```suggestion
     "outDir": "./dist",                              

```

**Review by LinaYahya — COMMENTED** [review]
Good Job Ahmed,
Just remove test folder outside server and create them as a js files,
and install jest and create a fake test,

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**